bcda/auth/middleware.go

```diff
@@ -63,25 +65,41 @@ func RequireTokenAuth(next http.Handler) http.Handler {
 	})
 }
 
-func RequireTokenACOMatch(next http.Handler) http.Handler {
+func RequireTokenJobMatch(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		jobID := chi.URLParam(r, "jobId")
 		token := r.Context().Value("token").(*jwt.Token)
+
+		db := database.GetGORMDbConnection()
+		defer database.Close(db)
+
+		i, err := strconv.Atoi(jobID)
+		if err != nil {
+			log.Error(err)
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, "", responseutils.Processing)
+			responseutils.WriteError(oo, w, http.StatusInternalServerError)
```

Comment:
400 or 404 would be more appropriate when the job ID in the URL is invalid.

Reply:
I stuck with 500 because there was a test that was expecting it.  I am happy to change the test though.

Reply:
+1 for a 404 here.

Reply:
all returns for this function are now 404